src/fields-present.js

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ * @licstart  The following is the entire license notice for the JavaScript code in this file.
+ *
+ * MARC record validators used in Melinda
+ *
+ * Copyright (C) 2014-2018 University Of Helsinki (The National Library Of Finland)
+ *
+ * This file is part of marc-record-validators-melinda
+ *
+ * marc-record-validators-melinda program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * marc-record-validators-melinda is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @licend  The above is the entire license notice
+ * for the JavaScript code in this file.
+ *
+ */
+
+'use strict';
+
+export default async function (tagPatterns) {
+	if (Array.isArray(tagPatterns)) {
+		return {
+			description:
+				'Checks whether the configured fields are present in the record',
+			validate: async record => ({
+				valid: isFieldPresent(record)
+			})
+		};
+	}
+	throw new Error('No tag pattern array provided');
+
+	function isFieldPresent(record) {
+		const values = record.fields.map(object => object.tag);
+		return values.every(value => isValid(value));
```

Comment:
This requires all tags to match the tag patterns. The functionality should be reversed: Each tag pattern should have at least one match, e.g.: `tagPatterns.every(pattern => record.fields.some(f => pattern.test(f.tag)))`

Reply:
refactored function